tests/Unit/NumericalAlgorithms/Spectral/Test_SwshDerivatives.cpp

```diff
@@ -0,0 +1,338 @@
+        make_not_null(&get(*computed_collocation).data()), modes, l_max,
+        number_of_radial_points, TestHelpers::spin_weighted_spherical_harmonic);
+  };
+  // Put the collocation information derived from the generated modes in the
+  // DataBox
+  db::mutate<TestTag<0, Spin1>>(make_not_null(&box),
+                                coefficients_to_analytic_collocation,
+                                expected_modes_spin_1);
+  db::mutate<TestTag<1, Spin2>>(make_not_null(&box),
+                                coefficients_to_analytic_collocation,
+                                expected_modes_spin_2);
+
+  // these could be packed into a variables, but the current test wouldn't be
+  // much shorter. If the test is expanded to verify more than four results at a
+  // time, using a `Variables` for the expected results is recommended for
+  // brevity.
+  ComplexDataVector expected_derivative_1_collocation_spin_1{
+      number_of_radial_points * number_of_swsh_collocation_points(l_max)};
+  ComplexDataVector expected_derivative_1_collocation_spin_2{
+      number_of_radial_points * number_of_swsh_collocation_points(l_max)};
+  ComplexDataVector expected_derivative_2_collocation_spin_1{
+      number_of_radial_points * number_of_swsh_collocation_points(l_max)};
+  ComplexDataVector expected_derivative_2_collocation_spin_2{
+      number_of_radial_points * number_of_swsh_collocation_points(l_max)};
+
+  TestHelpers::swsh_collocation_from_coefficients_and_basis_func<
+      Spin1, Representation>(
+      make_not_null(&expected_derivative_1_collocation_spin_1),
+      expected_modes_spin_1, l_max, number_of_radial_points,
+      TestHelpers::derivative_of_spin_weighted_spherical_harmonic<
+          DerivativeKind1>);
+
+  TestHelpers::swsh_collocation_from_coefficients_and_basis_func<
+      Spin2, Representation>(
+      make_not_null(&expected_derivative_1_collocation_spin_2),
+      expected_modes_spin_2, l_max, number_of_radial_points,
+      TestHelpers::derivative_of_spin_weighted_spherical_harmonic<
+          DerivativeKind1>);
+
+  TestHelpers::swsh_collocation_from_coefficients_and_basis_func<
+      Spin1, Representation>(
+      make_not_null(&expected_derivative_2_collocation_spin_1),
+      expected_modes_spin_1, l_max, number_of_radial_points,
+      TestHelpers::derivative_of_spin_weighted_spherical_harmonic<
+          DerivativeKind2>);
+
+  TestHelpers::swsh_collocation_from_coefficients_and_basis_func<
+      Spin2, Representation>(
+      make_not_null(&expected_derivative_2_collocation_spin_2),
+      expected_modes_spin_2, l_max, number_of_radial_points,
+      TestHelpers::derivative_of_spin_weighted_spherical_harmonic<
+          DerivativeKind2>);
+
+  // the actual derivative call
+  db::mutate_apply<SwshDerivatives<derivative_tag_list, Representation>>(
+      make_not_null(&box));
+
+  Approx swsh_approx =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e6)
```

Comment:
I'm surprised by how large the tolerance is. Could you add a comment as to why? I think in the Jacobi polynomial tests we make sure we can differentiate functions exactly, so if you're not doing that having a comment explaining it would be great.